src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -433,6 +532,342 @@ def can_create_clusterrolebindings(configuration):
+            try:
+                apd = subprocess.check_output(subprocess_cmd, universal_newlines=True)
+            except subprocess.CalledProcessError as ex:
+                logger.debug(f"Exception while running {subprocess_cmd}: {ex.returncode}, {ex.output}. Retrying...")
+                continue
+
+            apd_lines = apd.splitlines()
+            if apd_lines and 'No resources found' in apd_lines[0]:
+                apd_lines.pop(0)
+
+            print("Got {} diagnostic results for {} ready nodes{}\r".format(len(apd_lines),
+                                                                            len(ready_nodes),
+                                                                            '.' * retry), end='')
+            if len(apd_lines) < len(ready_nodes):
+                time.sleep(3)
+            else:
+                apds_created = True
+                print()
+        else:
+            for node_name in ready_nodes:
+                if ready_nodes[node_name]:
+                    continue
+                apdName = "aks-periscope-diagnostic-" + node_name
+                try:
+                    subprocess_cmd = kubectl_prior + ["get", "apd", apdName, "-n", "aks-periscope", "-o=jsonpath={.spec.networkconfig}"]
+                    network_config = subprocess.check_output(
+                        subprocess_cmd,
+                        universal_newlines=True)
+                    logger.debug('Dns status for node %s is %s', node_name, network_config)
+                    subprocess_cmd = kubectl_prior + ["get", "apd", apdName, "-n", "aks-periscope", "-o=jsonpath={.spec.networkoutbound}"]
+                    network_status = subprocess.check_output(
+                        subprocess_cmd,
+                        universal_newlines=True)
+                    logger.debug('Network status for node %s is %s', node_name, network_status)
+
+                    if not network_config or not network_status:
+                        print("The diagnostics information for node {} is not ready yet. "
+                              "Will try again in 10 seconds.".format(node_name))
+                        time.sleep(10)
+                        break
+
+                    network_config_array += json.loads('[' + network_config + ']')
+                    network_status_object = json.loads(network_status)
+                    network_status_array += format_diag_status(network_status_object)
+                    ready_nodes[node_name] = True
+                except subprocess.CalledProcessError as err:
+                    raise CLIInternalError(err.output)
+
+    print()
+    if network_config_array:
+        print("Below are the network configuration for each node: ")
+        print()
+        print(tabulate(network_config_array, headers="keys", tablefmt='simple'))
+        print()
+    else:
+        logger.warning("Could not get network config. "
```

Comment:
Let's enhance this part to make it clear to the user that there is an issue with network outbound connecitivity on his/her cluster

Reply:
Sure, @mallika-trivedi to suggest here.